application/client/src/app/service/session/dependencies/search/state.ts

```diff
@@ -116,4 +118,8 @@ export class State {
         this.subjects.get().active.emit(undefined);
         return this.filters();
     }
+
+    public showMatches(filter: IFilter) {
+        this.subjects.get().showMatches.emit(filter);
+    }
```

Comment:
no need to have this method

Reply:
and also this is logical mistake.
a) life-time of state = session
b) life-time of single listener of event `showMathces` is `app-views-search-input`-lifetime
if `app-views-search-input` doesn't exist, `showMathces` never will do any